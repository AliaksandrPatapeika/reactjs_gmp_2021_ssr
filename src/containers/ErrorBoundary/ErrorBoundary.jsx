/* eslint-disable no-console */
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';

import OopsText from '../../components/OopsText';

class ErrorBoundary extends Component {
	state = {
	  hasError: false
	};

	static getDerivedStateFromError() {
	  return {hasError: true};
	}

	componentDidCatch(error, errorInfo) {
	  console.log('error: ', error);
	  console.log('errorInfo: ', errorInfo);
	}

	render() {
	  const {hasError} = this.state;
	  const {children, errorMessage} = this.props;

	  return hasError || errorMessage ? <OopsText errorMessage={errorMessage} /> : children;
	}
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  errorMessage: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  errorMessage: state.movie.errorMessage
});

export default connect(mapStateToProps)(ErrorBoundary);
