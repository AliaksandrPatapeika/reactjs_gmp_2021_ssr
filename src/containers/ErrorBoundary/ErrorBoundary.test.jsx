import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        errorMessage: 'Error message'
      }
    };
    const store = mockStore(initialState);
    const errorBoundary = renderer
      .create(
        <Provider store={store}>
          <ErrorBoundary>
            Test
          </ErrorBoundary>
        </Provider>
      )
      .toJSON();

    expect(errorBoundary).toMatchSnapshot();
  });
});
