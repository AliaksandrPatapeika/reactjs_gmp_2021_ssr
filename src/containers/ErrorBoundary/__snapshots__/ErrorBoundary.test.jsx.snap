// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders correctly 1`] = `
<div
  className="oopsTextContainer"
>
  <div
    className="oopsTextErrorMessage"
  >
    Error message
  </div>
  <div>
    Oops, something went wrong...
  </div>
  <div>
    We are doing our best to fix the issue
  </div>
</div>
`;
